src/main/java/com/conveyal/r5/analyst/TravelTimeSurfaceComputer.java

```diff
@@ -0,0 +1,179 @@
+package com.conveyal.r5.analyst;
+
+import com.conveyal.r5.analyst.cluster.AccessGridWriter;
+import com.conveyal.r5.analyst.cluster.TravelTimeSurfaceRequest;
+import com.conveyal.r5.analyst.error.TaskError;
+import com.conveyal.r5.api.util.LegMode;
+import com.conveyal.r5.common.JsonUtilities;
+import com.conveyal.r5.profile.FastRaptorWorker;
+import com.conveyal.r5.profile.PerTargetPropagater;
+import com.conveyal.r5.profile.RaptorWorker;
+import com.conveyal.r5.profile.StreetMode;
+import com.conveyal.r5.publish.StaticComputer;
+import com.conveyal.r5.streets.LinkedPointSet;
+import com.conveyal.r5.streets.StreetRouter;
+import com.conveyal.r5.transit.TransportNetwork;
+import gnu.trove.iterator.TIntIntIterator;
+import gnu.trove.map.TIntIntMap;
+import org.apache.commons.math3.util.FastMath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.IntStream;
+
+/**
+ * This computes a travel time surface and returns it in access grid format, with one travel time for each minute/MC draw.
+ */
+public class TravelTimeSurfaceComputer {
+    private static final Logger LOG = LoggerFactory.getLogger(TravelTimeSurfaceComputer.class);
+    private static final WebMercatorGridPointSetCache pointSetCache = new WebMercatorGridPointSetCache();
+
+    public static final int PERCENTILE_STEP = 5;
+
+    public final TravelTimeSurfaceRequest request;
+    public final TransportNetwork network;
+
+    public TravelTimeSurfaceComputer (TravelTimeSurfaceRequest request, TransportNetwork network) {
+        this.request = request;
+        this.network = network;
+    }
+
+    public void write (OutputStream os) throws IOException {
+        StreetMode accessMode = LegMode.legModeSetToDominantStreetMode(request.request.accessModes);
+        StreetMode directMode = LegMode.legModeSetToDominantStreetMode(request.request.directModes);
+
+        int nSamples = 100 / PERCENTILE_STEP - 1;
```

Comment:
Since this always returns percentiles, I'd be in favor of changing this variable name to nPercentiles.